--- ifscheme.distrib	2017-02-07 16:52:33.000000000 +0000
+++ ifscheme	2017-07-14 19:07:43.938944551 +0000
@@ -1,10 +1,11 @@
 #!/bin/sh -e
+. /etc/libexec/update-patch-helper-5cj2dtd975yp5yah4ku3ll4ia
 # ifscheme changes network configuration schemes using ifup and ifdown
 
 # File locations
 OLD_IFSTATE=/etc/network/run/ifstate
 NEW_IFSTATE=/run/network/ifstate
-INTERFACES=/etc/network/interfaces
+INTERFACES=/etc/network/interfaces.d
 SCHEME=/run/network/scheme
 
 usage ()
@@ -20,7 +21,7 @@
 list()
 {
 	# Sed magic...
-	sed -ne "/^[[:space:]]*\(iface\|map\)[[:space:]].*-/{s:^[ ]*::;s:-: :;p;}" < $INTERFACES | cut -d' ' -f3 | sort -u
+	cat "$INTERFACES"/* | sed -ne "/^[[:space:]]*\(iface\|map\)[[:space:]].*-/{s:^[ ]*::;s:-: :;p;}" | cut -d' ' -f3 | sort -u
 
 	# Easier to read, slower :
 	#grep "^[[:space:]]*\(iface\|map\)[[:space:]].*-" $INTERFACES | sed -e"s/^[ ]*//" | sed -e"s/-/ /" | cut -d' ' -f3 | sort -u
@@ -84,7 +85,7 @@
 
 	# Does it look like this scheme exists ? Or does a wildcard scheme
 	# exist ?
-	if grep $quiet "^[[:space:]]*\(iface\|map\)[[:space:]].*-\($newscheme\|\*\)" $INTERFACES
+	if cat "$INTERFACES"/* | grep $quiet "^[[:space:]]*\(iface\|map\)[[:space:]].*-\($newscheme\|\*\)"
 	then
 		echo $newscheme > $SCHEME
 		echo "Scheme changed to $newscheme."
@@ -102,7 +103,7 @@
 		# First, extract liface names from explicit mapping lines
 		# in interface configuration file, either matching
 		# the old scheme or a wildcard.
-		old_mappings=$(sed -ne "/^[[:space:]]*map[[:space:]].*-\($oldscheme\|\*\)/{s:^[ ]*::p;}" < $INTERFACES | cut -d' ' -f 3)
+		old_mappings=$(cat "$INTERFACES"/* | sed -ne "/^[[:space:]]*map[[:space:]].*-\($oldscheme\|\*\)/{s:^[ ]*::p;}" | cut -d' ' -f 3)
 
 		# Second, extract from the state file any real interface
 		# which liface either match the mapping above or is
