--- pam-auth-update.disabled	2023-09-21 20:55:12.000000000 +0000
+++ pam-auth-update	2023-11-11 10:49:10.279393278 +0000
@@ -1,4 +1,21 @@
+#! /bin/sh
+
+# Local patch: Support /etc/pam-configs/* as an alternative to
+# /usr/share/pam-configs/*.
+#
+# Patch version 2023.315
+# Copyright (c) 2023 Guenther Brunthaler. All rights reserved.
+#
+# The patch applied to this script is free software.
+# Distribution is permitted under the terms of the GPLv3.
+
+ofile=/usr/sbin/pam-auth-update.disabled
+. /etc/libexec/update-patch-helper-5cj2dtd975yp5yah4ku3ll4ia
+
+exec /usr/bin/perl -x -- "$0" ${1+"$@"}
+
 #!/usr/bin/perl -w
+#line 19
 
 # pam-auth-update: update /etc/pam.d/common-* from /usr/share/pam-configs
 #
@@ -32,6 +49,7 @@
 my $capb=capb('backup escape');
 
 my $inputdir = '/usr/share/pam-configs';
+my $inputdir_alt = '/etc/pam-configs';
 my $template = 'libpam-runtime/profiles';
 my $errtemplate = 'libpam-runtime/conflicts';
 my $overridetemplate = 'libpam-runtime/override';
@@ -107,6 +125,9 @@
 	}
 }
 
+sub process_inputdir($) {
+	my($inputdir) = @_;
+
 opendir(DIR, $inputdir) || die "could not open config directory: $!";
 while (my $profile = readdir(DIR)) {
 	next if ($profile eq '.' || $profile eq '..' || $profile =~ m/~$/ || $profile =~ m/^#.+#$/);
@@ -117,6 +138,11 @@
 }
 closedir DIR;
 
+}
+
+process_inputdir($inputdir);
+process_inputdir($inputdir_alt);
+
 $priority = 'medium' if ($package);
 
 x_loadtemplatefile('/var/lib/dpkg/info/libpam-runtime.templates','libpam-runtime');
